scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala

```diff
@@ -206,6 +219,33 @@ object BigQueryTable {
     BigQueryTable(beam.BigQueryHelpers.toTableSpec(table))
 }
 
+/**
+ * Get an IO for a BigQuery table using the storage API.
+ */
+final case class BigQueryStorage(tableSpec: String) extends BigQueryIO[TableRow] {
+  override type ReadP = BigQueryStorage.ReadParam
+  override type WriteP = Nothing // ReadOnly
+
+  override protected def read(sc: ScioContext, params: ReadP): SCollection[TableRow] =
+    Reads.bqReadStorage(sc)(beam.BigQueryIO.readTableRows(),
+                            tableSpec,
+                            params.selectFields,
+                            params.rowRestriction)
+
+  override protected def write(data: SCollection[TableRow], params: WriteP): Tap[TableRow] =
+    throw new IllegalStateException("BigQueryStorage is read-only")
+
+  // FIXME: implement this
+  override def tap(read: ReadP): Tap[TableRow] = ???
```

Comment:
Probably not needed since the `IO` is read-only. Plus don't really want to port Beam IO internal conversion logic over.